de-lib/src/main/java/org/iplantc/de/apps/client/views/details/AppDetailsViewImpl.java

```diff
@@ -230,22 +238,84 @@ public void setBaseDebugId(final String baseId) {
         }
+
+        Scheduler.get().scheduleFinally(new Scheduler.ScheduledCommand() {
+
+            @Override
+            public void execute() {
+                final JavaScriptObject presenterShim = getPresenterShim(AppDetailsViewImpl.this);
+                final JavaScriptObject appearanceShim = getAppDetailsAppearanceShim(appearance);
+                final Splittable appJson = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(app));
+
+                renderToolDetails(toolsContainer.getId(), appearanceShim, appJson);
+                renderCategoryTree(hierarchyWidget.getElement().getId(), appJson, presenterShim, appearanceShim);
+            }
+        });
+    }
+
+    public static native void renderCopyTextArea(String elementID, String btnText, String textToCopy) /*-{
+        $wnd.CyVerseReactComponents.renderCopyTextArea(elementID, btnText, textToCopy);
+    }-*/;
+
+    public static native void renderToolDetails(String elementID, JavaScriptObject appearance, Splittable app) /*-{
+        $wnd.CyVerseReactComponents.renderToolDetails(elementID, appearance, app);
+    }-*/;
+
+    public static native void renderCategoryTree(String elementID, Splittable app, JavaScriptObject presenter, JavaScriptObject appearance) /*-{
+        $wnd.CyVerseReactComponents.renderCategoryTree(elementID, app, presenter, appearance);
+    }-*/;
+
+    public static native JavaScriptObject getAppDetailsAppearanceShim(AppDetailsView.AppDetailsAppearance appearance) /*-{
+        var css = appearance.@org.iplantc.de.apps.client.AppDetailsView.AppDetailsAppearance::css()();
+
+        return {
+            css: function() {
+                return {
+                    label: css.@org.iplantc.de.apps.client.AppDetailsView.AppDetailsAppearance.AppDetailsStyle::label(),
+                    value: css.@org.iplantc.de.apps.client.AppDetailsView.AppDetailsAppearance.AppDetailsStyle::value()
+                };
+            },
+            detailsLabel:         function() { return appearance.@org.iplantc.de.apps.client.AppDetailsView.AppDetailsAppearance::detailsLabel()(); },
+            toolNameLabel:        function() { return appearance.@org.iplantc.de.apps.client.AppDetailsView.AppDetailsAppearance::toolNameLabel()(); },
+            descriptionLabel:     function() { return appearance.@org.iplantc.de.apps.client.AppDetailsView.AppDetailsAppearance::descriptionLabel()(); },
+            toolPathLabel:        function() { return appearance.@org.iplantc.de.apps.client.AppDetailsView.AppDetailsAppearance::toolPathLabel()(); },
+            toolVersionLabel:     function() { return appearance.@org.iplantc.de.apps.client.AppDetailsView.AppDetailsAppearance::toolVersionLabel()(); },
+            toolAttributionLabel: function() { return appearance.@org.iplantc.de.apps.client.AppDetailsView.AppDetailsAppearance::toolAttributionLabel()(); }
+        };
+    }-*/;
+
+    public static native JavaScriptObject getPresenterShim(AppDetailsView presenter) /*-{
```

Comment:
These `shim` methods box the presenter and appearance method calls into a `JavaScriptObject` that can be passed to the React component, and mimics GWT 2.8's JSInterop.

If we upgrade to GWT 2.8, then I think the presenter and appearance interfaces can be annotated as a `@JsType`, then they can be passed directly to the React components without a `JavaScriptObject` shim.